src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -330,7 +330,14 @@ protected virtual (IBlockProcessor.IBlockTransactionsExecutor, IWorldState) GetO
         if (!block.IsGenesis && block.Transactions.Length != 0 && ExecutionTracer.IsTracingAccessWitness)
         {
             var gasWitness = new VerkleExecWitness(NullLogManager.Instance);
-            gasWitness.AccessForGasBeneficiary(block.Header.GasBeneficiary!);
+            var gasAvailable = block.GasLimit - block.GasUsed;
+            var gasBefore = gasAvailable;
+            if (!gasWitness.AccessForGasBeneficiary(block.Header.GasBeneficiary!, ref gasAvailable))
+            {
+                throw new OutOfGasException();
```

Comment:
there was no gas charge here - we just add witness here, without charging the gas

Reply:
fixed